src/app/Shared/Services/Api.service.tsx

```diff
@@ -866,7 +866,8 @@ export class ApiService {
     apiVersion: ApiVersion,
     path: string,
     config?: RequestInit,
-    suppressNotifications = false
+    suppressNotifications = false,
+    skipStatusCheck = false
```

Comment:
What if instead of `skipStatusCheck = false` this was `errorHandler?: Function`?

Further down,

```typescript
if (!errorHandler) {
    return this.handleError<Response>(err, req, suppressNotifications);
} else {
    errorHandler.apply(err, req);
}
```

Reply:
Yeh I was thinking about this but I decided to just throw the error and handle it outside sendRequest like below.

```typescript
testTarget(connectUrl: string, username?: string, password?: string) {
  return this.sendRequest(...).pipe(catchError(error) => { /* do something */ });
}
```

Just a thought to keep it clean and as close to "common practice". What do you think?

Reply:
Sure, that makes sense too.